addons/mass_mailing/migrations/13.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,61 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ('mail.mail.statistics', 'mailing.trace'),
+    ('mail.mass_mailing', 'mailing.mailing'),
+    ('mail.mass_mailing.contact', 'mailing.contact'),
+    ('mail.mass_mailing.list', 'mailing.list'),
+    ('mail.mass_mailing.list_contact_rel', 'mailing.contact.subscription'),
+]
+
+_table_renames = [
+    ('mail_mail_statistics', 'mailing_trace'),
+    ('mail_mass_mailing', 'mailing_mailing'),
+    ('mail_mass_mailing_contact', 'mailing_contact'),
+    ('mail_mass_mailing_list', 'mailing_list'),
+    # many2many tables
+    ('mail_mass_mailing_list_contact_rel', 'mailing_contact_list_rel'),
+    ('mail_mass_mailing_contact_res_partner_category_rel', 'mailing_contact_res_partner_category_rel'),
+    ('mail_mass_mailing_contact_list_rel', 'mailing_contact_list_rel'),
+    ('mail_mass_mailing_tag_rel', 'utm_tag_rel'),
+    # removed tables
+    ('mail_mass_mailing_campaign', None),
```

Comment:
no need to remove

Reply:
I want to copy that table actually to merge in utm_campain in post-migration.py

Reply:
but there is no need to copy. this table is not removed in the migration. you can make us of it for reference, no need a second one.